rest/route/agent.go

```diff
@@ -317,6 +317,65 @@ func (h *newPushHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(newPushLog)
 }
 
+// POST /task/{task_id}/restart
+type markTaskForRestartHandler struct {
+	taskID string
+}
+
+func makeMarkTaskForRestart() gimlet.RouteHandler {
+	return &markTaskForRestartHandler{}
+}
+
+func (h *markTaskForRestartHandler) Factory() gimlet.RouteHandler {
+	return &markTaskForRestartHandler{}
+}
+
+func (h *markTaskForRestartHandler) Parse(ctx context.Context, r *http.Request) error {
+	if h.taskID = gimlet.GetVars(r)["task_id"]; h.taskID == "" {
+		return errors.New("missing task ID")
+	}
+	return nil
+}
+
+func (h *markTaskForRestartHandler) Run(ctx context.Context) gimlet.Responder {
+	t, err := task.FindOneId(h.taskID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding task '%s'", h.taskID))
+	}
+	if t == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task '%s' not found", h.taskID),
+		})
+	}
+	taskToRestart := t
+	if t.IsPartOfDisplay() {
+		dt, err := t.GetDisplayTask()
+		if err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "getting display task for execution task '%s'", h.taskID))
+		}
+		taskToRestart = dt
+	}
+	// If the task is a display task that has already been marked for an automatic restart
+	// by another execution task, we don't want to mark it for a restart again nor do we
+	// want to error out since the display task has not been automatically restarted yet.
+	if taskToRestart.IsAutomaticRestart {
+		return gimlet.NewJSONResponse(struct{}{})
+	}
+
+	if taskToRestart.NumAutomaticRestarts < evergreen.MaxAutomaticRestarts {
```

Comment:
nit: I'd reorder this to avoid the nested if statement, and having the error away from the if statement:

```
if we've exceeded restarts {
    return error 
}
if SetResetWhenFinished errors {
   return error
}
return no error
```

Reply:
Done.